### PR TITLE
[ADD] hr : Added functionality to calculate department, Employee and organization Age.

### PR DESCRIPTION
**Pull request by Solufyin**
[ADD] hr: Added functionality to calculate department, employee and organization age.

Hello @revant, @rmehta,

We Added functionality to calculate department, employee and organization age.

employee_age

Calculates average age and total age of employees of the particular department and the total number of employees in particular department. see the below screenshot.

department_age

Calculates average age and total age of employees of the organization and the total number of employees in the organization. see the below screenshot.

company_age

Thanks

### CODE REVIEW

File: erpnext/hooks.py [hunk 1]
```diff
@@ -211,6 +211,10 @@
 	},
 	('Sales Invoice', 'Purchase Invoice', 'Delivery Note'): {
 		'validate': 'erpnext.regional.india.utils.set_place_of_supply'
+	},
+	('Employee'): {
+		"on_update": [
+			"erpnext.hr.doctype.employee.employee.age_calculation"]
```
Comment (rmehta) [inline erpnext/hooks.py]: this should be called from the controller itself!

File: erpnext/hr/doctype/employee/employee.py [hunk 2]
```diff
@@ -209,6 +211,80 @@ def update_user_permissions(doc, method):
...
+		                    "total_number_of_employee", no_of_emp)
+		frappe.db.set_value(doctype, str(name), "total_age", total_age)
+		frappe.db.set_value(doctype, str(name), "average_age", average_age)
+
+def age_calculation(doc, method):
+	# called vis hooks.py doc event
+	department = doc.department or False
+	company = doc.company or False
+	if company:
+		company_emps = get_employees_by_filter({'company': company,
```
Comment (rmehta) [inline erpnext/hr/doctype/employee/employee.py]: Just use `frappe.get_all` !

```
frappe.get_all('Employee', dict(company = company, status = 'Active'), ['name', 'employee_age'])
```

**Comment by rmehta** [comment]
Needs documentation an tests!

@Solufyin Appreciate your team taking lead in contributions 👍  Please make sure you read this carefully

https://github.com/frappe/erpnext/wiki/Contribution-Guidelines

Lets build awesome features :)

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by Solufyin** [comment]
Hello @rmehta and @revant 

Thanks for appreciation.
We did improved code as per your suggestion.

Thanks.

**Comment by nabinhait** [comment]
@Solufyin Thanks for the PR.

As per the implementation, the age values will be update on save of Employee records or on change of date-of-birth field's value. So, at any point of time, it will not give accurate age of department / company employees. If we maintain those age fields, we have to calculate and update the values regularly via scheduler.

But in my opinion, we don't need those fields inside the department / company at all. There should be a **Report** to show age of employees based on department / company.

Hence, closing this PR.

**Pull request closed without merging**